src/logger/utils.ts

```diff
@@ -0,0 +1,33 @@
+import * as vscode from "vscode";
+
+type NotificationAction = () => Thenable<unknown> | Promise<void> | void;
+
+/**
+ * Shows an information notification with a button that executes a custom action when clicked.
+ * @param {string} infoMessage - The information message to display.
+ * @param {string} buttonLabel - The label for the button.
+ * @param {NotificationAction} action - The action to perform when the button is clicked.
+ * @returns {Promise<void>} - A promise that resolves when the notification is shown and handled.
+ */
+export async function showInfoNotificationWithAction(
+  infoMessage: string,
+  buttonLabel: string,
+  action: NotificationAction
+): Promise<void> {
+  const selectedOption = await vscode.window.showInformationMessage(
+    infoMessage,
+    buttonLabel
+  );
+
+  if (selectedOption === buttonLabel) {
+    await Promise.resolve(action());
+  }
+}
+
+// Helper functions for common actions
+export function executeCommand(
+  commandId: string,
+  ...args: any[]
+): NotificationAction {
+  return () => vscode.commands.executeCommand(commandId, ...args);
```

Comment:
I feel this function is unnecessary just doing `() => vscode.commands.executeCommand(commandId, ...args)` as argument or do `as NotificationAction` in case linter complains.